File: procon7/c/q1/Main.c
#include <stdio.h>

int main(int argc, char** argv) {
	const char* S = "ABCHJK";
	const char* L = "ABCDEFGHJK";

	char x;
	int a;
	int b;

	char col;

	fscanf(stdin, "%c %d %d", &x, &a, &b);

	if (x == 'S') {
		col = S[b - 1];
	} else {
		col = L[b - 1];
	}

	fprintf(stdout, "%d%c\n", a, col);

	return 0;
}

File: procon7/java/q1/Main.java
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class Main {
	static Map<String, String> map = new HashMap<>();

	static {
		map.put("S", "ABCHJK");
		map.put("L", "ABCDEFGHJK");
	}

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);

		String x = sc.next();
		int a = sc.nextInt();
		int b = sc.nextInt();
		char col = map.get(x).charAt(b - 1);
		System.out.println("" + a + col);
	}
}
